src/debugger/communication/server_controller.ts

```diff
@@ -0,0 +1,332 @@
+				let frames: Map<string, any>[] = params;
+				this.trigger_breakpoint(
+					frames.map(sf => {
+						return {
+							id: sf.get("id"),
+							file: sf.get("file"),
+							function: sf.get("function"),
+							line: sf.get("line")
+						};
+					})
+				);
+			})
+		);
+
+		this.builder.register_command(
+			new Command("output", params => {
+				params.forEach(line => {
+					this.output_channel?.appendLine(line);
+				});
+			})
+		);
+
+		this.builder.register_command(
+			new Command("error", params => {
+				params.forEach(param => {});
+			})
+		);
+
+		this.builder.register_command(new Command("performance", params => {}));
+
+		this.builder.register_command(
+			new Command("message:inspect_object", params => {
+				let id = params[0];
+				let class_name = params[1];
+				let properties = params[2];
+
+				let cb = this.inspected_callbacks.get(id);
+				if (cb) {
+					cb(class_name, properties);
+					this.inspected_callbacks.delete(id);
+				}
+			})
+		);
+
+		this.builder.register_command(
+			new Command("stack_frame_vars", params => {
+				let locals: any[] = [];
+				let members: any[] = [];
+				let globals: any[] = [];
+
+				let local_count = (params[0] as number) * 2;
+				let member_count = params[1 + local_count] * 2;
+				let global_count = params[2 + local_count + member_count] * 2;
+
+				if (local_count > 0) {
+					locals = params.slice(1, 1 + local_count);
+				}
+				if (member_count > 0) {
+					members = params.slice(
+						2 + local_count,
+						2 + local_count + member_count
+					);
+				}
+				if (global_count > 0) {
+					globals = params.slice(
+						3 + local_count + member_count,
+						3 + local_count + member_count + global_count
+					);
+				}
+
+				this.pumpScope(
+					{
+						locals: locals,
+						members: members,
+						globals: globals
+					},
+					project_path
+				);
+			})
+		);
+
+		this.server = net.createServer(connection => {
+			this.connection = connection;
+			this.godot_commands?.set_connection(connection);
+
+			connection.on("data", buffer => {
+				if (!this.parser || !this.builder) {
+					return;
+				}
+
+				let len = buffer.byteLength;
+				let offset = 0;
+				do {
+					let data = this.parser.get_buffer_dataset(buffer, offset);
+					let data_offset = data[0] as number;
+					offset += data_offset;
+					len -= data_offset;
+					this.builder.parse_data(data.slice(1));
+				} while (len > 0);
+			});
+
+			connection.on("close", hadError => {
+				if (hadError) {
+					this.send_event("terminated");
+				}
+			});
+
+			connection.on("end", () => {
+				this.send_event("terminated");
+			});
+
+			connection.on("error", error => {
+				console.error(error);
+			});
+
+			connection.on("drain", () => {
+				connection.resume();
+				this.godot_commands?.set_can_write(true);
+			});
+		});
+
+		this.server?.listen(port, address);
+
+		if (launch_game_instance) {
+			let executable_line = `godot --path ${project_path} --remote-debug ${address}:${port}`;
```

Comment:
We'd better using the godot editor of the plugin configuration instead of on the path.
For example I installed offecial version on the system path as 'godot'. But I most case I'm going to launch the binary compiled with my modules.